core/src/main/java/feign/Response.java

```diff
@@ -45,7 +45,8 @@
   private final Request request;
 
   private Response(Builder builder) {
-    checkState(builder.status >= 200, "Invalid status code: %s", builder.status);
+    checkState(builder.status >= 100 && builder.status < 600, "Invalid status code: %s",
```

Comment:
Personally, I don't see any reason to limit the response code at all. If somebody has a custom server that returns `65536` (or even `-654321`) then why not return the status code that was actually returned?

See [this stackoverflow post](https://stackoverflow.com/a/43094430/2167531) and [this random web discussion](https://community.softwaregrp.com/t5/Application-Perf-Mgmt-BAC-BSM/Possible-negative-status-codes-in-SiteScope/m-p/265881/highlight/true#M9346).

Reply:
That said, I have no objections to this PR.

Reply:
See also [this blog post](https://pokeinthe.io/2017/09/14/http-status-code-handling/).

Reply:
@rage-shadowman Thanks for sharing the blog post! I'd be more than happy to make the change so that it doesn't limit the status code. I'll admit it feels a little awkward to be enforcing it here, and not allowing users to have a choice in the matter.

Reply:
@maly7 Are you considering making additional updates to the PR to address @rage-shadowman's feedback?

Reply:
Yes, I'll have those changes pushed up tonight